pkg/cedar/core/authority_to_operate.go

```diff
@@ -0,0 +1,126 @@
+package cedarcore
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/guregu/null"
+	"github.com/guregu/null/zero"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
+
+	apiauthority "github.com/cmsgov/easi-app/pkg/cedar/core/gen/client/authority_to_operate"
+)
+
+// GetAuthorityToOperateOptionalParams represents the optional parameters that can be used to filter ATO information when searching through the CEDAR API
+type GetAuthorityToOperateOptionalParams struct {
+	CedarSystemID                             null.String
+	ContainsPersonallyIdentifiableInformation null.Bool
+	DispositionDateAfter                      *time.Time
+	DispositionDateBefore                     *time.Time
+	FismaSystemAcronym                        null.String
+	IsProtectedHealthInformation              null.Bool
+	SystemID                                  null.String
+	TlcPhase                                  null.String
+	UUID                                      null.String
+}
+
+// GetAuthorityToOperate makes a GET call to the /authority_to_operate endpoint
+func (c *Client) GetAuthorityToOperate(ctx context.Context, optionalParams *GetAuthorityToOperateOptionalParams) ([]*models.CedarAuthorityToOperate, error) {
```

Comment:
Happy to sync on this, since I think you might have been working on this at the same time as @jstrang-oddball - but our CEDAR Core methods now (and probably should for this too):
- Expect the "CedarSystemID" to be a required parameter (maybe scoot it from the optional params into a top-level required one?) that is the "version agnostic" ID for the system
- Use that ID to get the system from the cache, and then use the cached `system.VersionID` to make the query (since all these endpoints expect a version-specific ID, not an ICTObjectID)


An example of this can be seen in `GetRolesBySystem`, source code here for getting the system [here](https://github.com/CMSgov/easi-app/blob/bede23042e21b3b6319f9126fd0bd43791b43655/pkg/cedar/core/role.go#L46-L49), and setting the appropriate ID [here](https://github.com/CMSgov/easi-app/blob/bede23042e21b3b6319f9126fd0bd43791b43655/pkg/cedar/core/role.go#L54)